### DISCUSSION TITLE
How do I make this Child show as a footer?

### DISCUSSION THREAD

**Discussion by vbmark** [AllenDang/giu#648] [Q&A] [answered]
```go
func loop() {
	g.SingleWindow().Layout(g.Layout{
		g.Row(
			g.Child().Size(250, -1).Layout(g.Layout{
				g.Label("Navigation"),
			}),
			g.Child().Size(-1, -1).Layout(g.Layout{
				g.Label("Main Area"),
			}),
		),
		g.Row(
			g.Child().Size(-1, 30).Layout(g.Layout{
				g.Label("Footer"),
			}),
		),
	})
}
```
You can see a couple of pixels of the footer at the bottom of the screenshot.

![Screenshot](https://user-images.githubusercontent.com/89524/221665181-4801f5b4-a610-41c6-906f-af5cfc9c1d57.png)

Thank you.

**Comment by gucio321** [comment]
you need to do it this way:
1. get available height
2. get H spacing (`giu.GetItemSpacing()` )
3. think of the size of your footer: you have two options here:
  - make it constant (e.g. 200 or something)
  - or percentage (I think it'd be better)
5. set size of your two childs to (for percentage value) `(availableH-itemSpacingH)*(1-footerPercentage)`
6. and render your footer with the H size equal to `(availableH-itemSpacingH)*footerPercentage`

should work ;-)

**Reply by vbmark** [reply]
Hi, thank you for responding.

How do you get the height?  I tried this:

```go
wnd := g.Window("wnd")
w, h := wnd.CurrentSize()
println(w, h)
```
But it prints:
+0.000000e+000 +0.000000e+000

**Reply by gucio321** [reply]
```go
giu.Window("wnd").Layout(
   giu.Custom(func() {
      _, availableH := giu.GetAvailableRegion()
     _, spacingH := giu.GetItemSpacing()
     giu.Layout{
         /* render your childs*/
      }.Build()
   }),
)
```

**Reply by vbmark** [reply]
The values don't seem right:

![Screenshot - 2](https://user-images.githubusercontent.com/89524/221690558-65b75fdc-8783-42c3-9a1d-b6f0adfd0f35.png)

**Reply by gucio321** [reply]
@vbmark thats probably because your winodw is of the minimal size by default.
Here is the code that works for me:
```go
package main

import "github.com/AllenDang/giu"

func loop() {
	giu.Window("wnd").Layout(
		giu.Custom(func() {
			const footerPercentage = 0.2
			_, availableH := giu.GetAvailableRegion()
			_, itemSpacingH := giu.GetItemSpacing()
			giu.Layout{
				giu.Child().Layout(giu.Label("your layout")).Size(-1, (availableH-itemSpacingH)*(1-footerPercentage)),
				giu.Child().Layout(giu.Label("footer")).Size(-1, (availableH-itemSpacingH)*footerPercentage),
			}.Build()
		}),
	)
}

func main() {
	wnd := giu.NewMasterWindow("How do I make this Child show as a footer? #648", 640, 480, 0)
	wnd.Run(loop)
}
```

**Reply by gucio321** [reply]
If you want to set default size of your window, use the `Size` method.
Se also `giu.WindowFlagsNoResize` and `giu.StyleVarWindowMinSize`

**Reply by vbmark** [reply]
Please look at my code and my screenshot.  I expect three child windows:

Row One of two childs as columns:
-- The left column, named "Navigation", is fixed width at 250.
-- The right column, named "Main Area", fills the rest of the horizontal space.

Row Two, a single child named "Footer", fixed at a height of 30, is below Row One.

You can see some of the top of the child in row two if you look real close.

I can scroll and see row two, but I should not have to scroll.  The footer should always be displaying at the bottom regardless of how the window is resized.

**Comment by AllenDang** [comment]
```go
package main

import (
	g "github.com/AllenDang/giu"
)

func loop() {
	g.SingleWindow().Layout(
		g.Child().ID("Container").Layout(
			g.Label("Add anything here"),
		).Size(g.Auto, -40),
		g.Child().ID("Footer").Layout(
			g.Row(
				g.Label("Footer"),
				g.Label("status bar"),
			),
		).Size(g.Auto, g.Auto).Flags(g.WindowFlagsNoScrollbar),
	)
}

func main() {
	wnd := g.NewMasterWindow("Hello world", 400, 200, 0)
	wnd.Run(loop)
}
```
![footer](https://user-images.githubusercontent.com/18550/222031308-5328db10-b792-4537-a8fb-cd6e5356f9c9.gif)

**Reply by vbmark** [reply]
I appreciate the reply but it's not addressing the issue that I am stating.

Please look at my code and my screenshot. I expect three child windows:

Row One of two childs as columns:
-- The left column, named "Navigation", is fixed width at 250.
-- The right column, named "Main Area", fills the rest of the horizontal space.

Row Two, a single child named "Footer", fixed at a height of 30, is below Row One.

You can see some of the top of the child in row two if you look real close.

I can scroll and see row two, but I should not have to scroll. The footer should always be displaying at the bottom regardless of how the window is resized.

**Reply by AllenDang** [reply]
No matter what layout you are trying to make, the technice is the same.

**Reply by AllenDang** [reply, marked as answer]
Here is the demo code using same tech.
```go
package main

import (
	g "github.com/AllenDang/giu"
)

func loop() {
	g.SingleWindow().Layout(
		g.Row(
			g.Child().ID("Navigator").Size(230, -40).Layout(
				g.Label("Add anything here"),
			),
			g.Child().ID("Container").Size(g.Auto, -40).Layout(
				g.Label("Container"),
			),
		),
		g.Child().ID("Footer").Layout(
			g.Row(
				g.Label("Footer"),
				g.Label("status bar"),
			),
		).Size(g.Auto, g.Auto).Flags(g.WindowFlagsNoScrollbar),
	)
}

func main() {
	wnd := g.NewMasterWindow("Hello world", 800, 500, 0)
	wnd.Run(loop)
}
```
![footer](https://user-images.githubusercontent.com/18550/222317758-f130a429-d6ed-46ae-afce-0e794bd7c144.gif)

**Reply by vbmark** [reply]
That is awesome! Thank you!
